environments/gcp/flyte-core/iam.tf

```diff
@@ -118,6 +118,15 @@ resource "google_project_iam_binding" "flyteworkers-binding" {
   
 }
 
+#This particular binding enables the necessary permissions to build and push images to an Artifact Registry.
```

Comment:
Why do the flyte workers need permission to write images into ar? Can there be a situation where images are attempted to be built in the cluster when using `ImageSpec`? Ever since GKE replaced docker in ContainerOS with containerd, it's not easily possible to build images in a GKE cluster ..

I would expect that images are always built either in CI or locally on the users notebook/dev machine, meaning that their user account needs the permission to push, not the flyte workers service account.

Reply:
Not needed. This was being used both as the SA where users could get a token from to push images and for the Pods to pull them. This is not good. I just tested and changed the structure so there's a GSA that users/CI can leverage to obtain a token, auth to the docker repo and push images. The workers GSA is granted the `artifactregistry.reader` role to pull images.